> Actually, that PR is the one that broke this, right?

That's the PR that would break this once the changes are released, yes. (Hence #105 still being a draft). Just to confirm, were you building both image_common and image_transport_plugins from source?

The original problem that https://github.com/ros-perception/image_common/pull/249 solved was the error message that showed up at run time:
```
[ERROR] [1659558188.442971154] [rqt_gui_cpp_node_1318445]: SubscriberPlugin::subscribeImpl with five arguments has not been overridden
```
showed up when you called advertiseImpl without options, so there was a PR that hacked around this (https://github.com/ros-perception/image_common/pull/243). In https://github.com/ros-perception/image_common/pull/243#issuecomment-1186926280, I suggested deprecating image transports that don't handle options correctly.

--- a/compressed_depth_image_transport/include/compressed_depth_image_transport/compressed_depth_publisher.h
+++ b/compressed_depth_image_transport/include/compressed_depth_image_transport/compressed_depth_publisher.h
@@ -50,12 +50,13 @@ public:
 
 protected:
   // Overridden to set up reconfigure server
-  virtual void advertiseImpl(
+  void advertiseImpl(
           rclcpp::Node * node,
           const std::string &base_topic,
-          rmw_qos_profile_t custom_qos) override final;
+          rmw_qos_profile_t custom_qos,
+          rclcpp::PublisherOptions options) override final;
 
-  virtual void publish(const sensor_msgs::msg::Image& message,
+  void publish(const sensor_msgs::msg::Image& message,
                        const PublishFn& publish_fn) const override final;
 
   struct Config {

--- a/compressed_depth_image_transport/src/compressed_depth_publisher.cpp
+++ b/compressed_depth_image_transport/src/compressed_depth_publisher.cpp
@@ -60,10 +60,11 @@ namespace compressed_depth_image_transport
 void CompressedDepthPublisher::advertiseImpl(
   rclcpp::Node * node,
   const std::string& base_topic,
-  rmw_qos_profile_t custom_qos)
+  rmw_qos_profile_t custom_qos,
+  rclcpp::PublisherOptions options)
 {
   typedef image_transport::SimplePublisherPlugin<sensor_msgs::msg::CompressedImage> Base;
-  Base::advertiseImpl(node, base_topic, custom_qos);
+  Base::advertiseImpl(node, base_topic, custom_qos, options);
 
   node->get_parameter_or<int>("png_level", config_.png_level, kDefaultPngLevel);
   node->get_parameter_or<double>("depth_max", config_.depth_max, kDefaultDepthMax);

--- a/compressed_image_transport/include/compressed_image_transport/compressed_subscriber.h
+++ b/compressed_image_transport/include/compressed_image_transport/compressed_subscriber.h
@@ -55,12 +55,6 @@ public:
   }
 
 protected:
-  // Overridden to set up reconfigure server
-  void subscribeImpl(
-      rclcpp::Node * ,
-      const std::string& base_topic,
-      const Callback& callback,
-      rmw_qos_profile_t custom_qos) override;
 
   void subscribeImpl(
       rclcpp::Node * ,

--- a/compressed_image_transport/src/compressed_subscriber.cpp
+++ b/compressed_image_transport/src/compressed_subscriber.cpp
@@ -61,21 +61,12 @@ void CompressedSubscriber::subscribeImpl(
     rclcpp::Node * node,
     const std::string& base_topic,
     const Callback& callback,
-    rmw_qos_profile_t custom_qos)
-{
-  this->subscribeImpl(node, base_topic, callback, custom_qos, rclcpp::SubscriptionOptions{});
-}
-
-void CompressedSubscriber::subscribeImpl(
-    rclcpp::Node * node,
-    const std::string& base_topic,
-    const Callback& callback,
     rmw_qos_profile_t custom_qos,
     rclcpp::SubscriptionOptions options)
 {
     logger_ = node->get_logger();
     typedef image_transport::SimpleSubscriberPlugin<CompressedImage> Base;
-    Base::subscribeImplWithOptions(node, base_topic, callback, custom_qos, options);
+    Base::subscribeImpl(node, base_topic, callback, custom_qos, options);
     uint ns_len = node->get_effective_namespace().length();
     std::string param_base_name = base_topic.substr(ns_len);
     std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');

--- a/theora_image_transport/include/theora_image_transport/theora_subscriber.h
+++ b/theora_image_transport/include/theora_image_transport/theora_subscriber.h
@@ -51,12 +51,12 @@ public:
 
 protected:
   // Overridden to bump queue_size, otherwise we might lose headers
-  virtual void subscribeImpl(
+  void subscribeImpl(
     rclcpp::Node* node,
     const std::string &base_topic,
     const Callback & callback,
-    uint32_t queue_size,
-    rmw_qos_profile_t custom_qos);
+    rmw_qos_profile_t custom_qos,
+    rclcpp::SubscriptionOptions options) override;
 
   // The function that does the actual decompression and calls a user supplied
   // callback with the resulting image

--- a/theora_image_transport/src/theora_subscriber.cpp
+++ b/theora_image_transport/src/theora_subscriber.cpp
@@ -69,20 +69,13 @@ void TheoraSubscriber::subscribeImpl(
   rclcpp::Node * node,
   const std::string &base_topic,
   const Callback & callback,
-  uint32_t queue_size,
-  rmw_qos_profile_t custom_qos)
+  rmw_qos_profile_t custom_qos,
+  rclcpp::SubscriptionOptions options)
 {
   logger_ = node->get_logger();
-  // queue_size doesn't account for the 3 header packets, so we correct (with a little extra) here.
-  // ported this to ROS2 using the history policy that  determines how messages
-  // are saved until the message is taken by the reader.  KEEP_ALL saves all
-  // messages until they are taken.  KEEP_LAST enforces a limit on the number of
-  // messages that are saved, specified by the "depth" parameter.
-  custom_qos.history = rmw_qos_profile_default.history;
-  custom_qos.depth = queue_size + 4;
 
   typedef image_transport::SimpleSubscriberPlugin<theora_image_transport::msg::Packet> Base;
-  Base::subscribeImpl(node, base_topic, callback, custom_qos);
+  Base::subscribeImpl(node, base_topic, callback, custom_qos, options);
 }
 
 // TODO: port this check to ROS2 user events

